bar-stacked-diverging/parseData.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * General data munging functionality
+ */
+
+import * as d3 from 'd3';
+import loadData from '@financial-times/load-data';
+
+/**
+ * Parses data file and returns structured data
+ * @param  {String} url Path to CSV/TSV/JSON file
+ * @return {Object}     Object containing series names, value extent and raw data object
+ */
+export function load(url, options) { // eslint-disable-line
+    return loadData(url).then((result) => {
+        const data = result.data ? result.data : result;
+
+        const seriesNames = getSeriesNames(data.columns);
+
+        const rowValueExtents = [];
+        // Use the seriesNames array to calculate the minimum and max values in the dataset
+        const valueExtent = extentMulti(data, seriesNames, rowValueExtents);
+
+
+        // function that calculates the position of each rectangle in the stack
+        const getStacks = function getStacks(el, key) {
+            let posCumulative = 0;
+            let negCumulative = 0;
+            let baseX = 0;
+            let baseX1 = 0;
+            const stacks = seriesNames.map((name, i) => {
+                if (el[name] > 0) {
+                    baseX1 = posCumulative;
+                    posCumulative += (+el[name]);
+                    baseX = posCumulative;
+                }
+                if (el[name] < 0 && i > 0) {
+                    baseX1 = negCumulative;
+                    negCumulative += (+el[name]);
+                    baseX = negCumulative;
+                }
+                // set the first bar at minimum value position
+                if (i < 1) {
```

Comment:
Might be worth making the chain of `if` statements above into a series of if-else-if statements to make it a bit more intuitive (as stands it's possible `baseX1`, `posCumulative` and `baseX` are set multiple times in the same loop iteration, which makes it harder to figure out what's going on).